### ISSUE TITLE
type(vec2d) seems to have a function that treats it as 3D

### ISSUE THREAD

**Issue by Beliavsky** [jordan4ibanez/fglm#1]
In your code below for vector_2d.f90 should the function `constructor_raw_f64` have argument z?

```
  type(vec2d) function constructor_raw_f64(x,y,z) result(new_vec2d)
    implicit none

    real(c_double), intent(in), value :: x,y,z

    new_vec2d = [x,y,z]
  end function constructor_raw_f64


  type(vec2d) function constructor_array_f64(xyz_array) result(new_vec2d)
    implicit none

    real(c_double), dimension(2), intent(in) :: xyz_array

    new_vec2d = xyz_array(1:2)
  end function constructor_array_f64
```

**Comment by jordan4ibanez** [comment]
WHOOPS! LOL. Thanks for catching that, let me fix them.

**Comment by jordan4ibanez** [comment]
Fixed in a3477f7
